### PR TITLE
Feat: Add new public API for WorkflowRun which returns serialized values and artifacts

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
We need to have public API that returns serialized results and artifacts, as this is used by Workflow-driver-ray

# This PR's solution

Add New APIs

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_base/_api/_wf_run.py [hunk 1]
```diff
@@ -399,6 +400,38 @@ def get_results(self, wait: bool = False) -> t.Sequence[t.Any]:
 
         return results
 
+    def get_results_serialized(self, wait: bool = False) -> t.Sequence[WorkflowResult]:
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_base/_api/_wf_run.py:403]: **suggestion:** ‏wdty about adding a new keyword only argument to `get_results` instead of a new method? maybe `serialized: bool = False`

Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/_api/_wf_run.py:403]: I could do that - but Im not super into that idea. Why? Coz basically the function returns enterily different data structure. Meaning we would have function that returns either <any> (as whatever WF function reruns) OR sequence of workflowResults.

TLDR: I could do that but I prefer the way it is. maybe @alexjuda has any strong opinions?

Comment (alexjuda) [reply src/orquestra/sdk/_base/_api/_wf_run.py:403]: I thought about a flag like this, but I think having a separate method is better. It avoids ambiguity in the returned type, usually leads to a better code completion, and avoids an `if` in our code + `assert isinstance(result, ...)` on the caller side.

Comment (jamesclark-Zapata) [reply src/orquestra/sdk/_base/_api/_wf_run.py:403]: OK, how about using the `get_results_serialized` inside `get_results`? Just trying to minimise the number of locations we call the underlying runtime

Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/_api/_wf_run.py:403]: Absolutely makes sense, will do that

File: src/orquestra/sdk/_base/_api/_wf_run.py [hunk 2]
```diff
@@ -382,23 +379,62 @@ def get_results(self, wait: bool = False) -> t.Sequence[t.Any]:
...
+    def get_results(self, wait: bool = False) -> t.Sequence[t.Any]:
+        """
+        Retrieves workflow results, as returned by the workflow function.
+
+        A workflow function is expected to return task outputs
+        (ArtifactFutures) or constants (10, "hello", etc.). This method returns values
+        of these. The order is dictated by the return statement in the workflow
+        function, for example `return a, b, c` means this function returns (a, b, c).
+        See also:
+        https://refactored-disco-d576cb73.pages.github.io/docs/runtime/guides/workflow-syntax.html
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_api/_wf_run.py:395]: **nitpick**: inb4 I know it's been here before. This is an internal link. Let's use the public one:
```suggestion
        https://docs.orquestra.io/docs/core/sdk/guides/workflow-syntax.html/workflow-syntax.html
```

File: src/orquestra/sdk/_base/_api/_wf_run.py [hunk 3]
```diff
@@ -382,23 +379,62 @@ def get_results(self, wait: bool = False) -> t.Sequence[t.Any]:
...
+
+        Args:
+            wait:  whether or not to wait for workflow run completion.
+                   Uses the default options for waiting, use `wait_until_finished()` for
+                   more control.
+
+        Raises:
+            WorkflowRunNotFinished: when the workflow run has not finished and `wait` is
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/_api/_wf_run.py:403]: nitpick: rst syntax
```suggestion
            WorkflowRunNotFinished: when the workflow run has not finished and ``wait`` is
```

**Review by alexjuda — APPROVED** [review]
Added non-blocking nitpicks
